src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -104,12 +114,34 @@ private void CreateClient()
                 var credentials = new BasicAWSCredentials("dummy", "dummyKey");
                 this.ddbClient = new AmazonDynamoDBClient(credentials, new AmazonDynamoDBConfig { ServiceURL = this.service });
             }
+            else if (!string.IsNullOrEmpty(this.accessKey) && !string.IsNullOrEmpty(this.secretKey) && !string.IsNullOrEmpty(this.token))
+            {
+                // AWS DynamoDB instance (auth via explicit credentials and token)
+                var credentials = new SessionAWSCredentials(this.accessKey, this.secretKey, this.token);
+                this.ddbClient = new AmazonDynamoDBClient(credentials, new AmazonDynamoDBConfig {RegionEndpoint = AWSUtils.GetRegionEndpoint(this.service)});
+            }
             else if (!string.IsNullOrEmpty(this.accessKey) && !string.IsNullOrEmpty(this.secretKey))
             {
                 // AWS DynamoDB instance (auth via explicit credentials)
                 var credentials = new BasicAWSCredentials(this.accessKey, this.secretKey);
                 this.ddbClient = new AmazonDynamoDBClient(credentials, new AmazonDynamoDBConfig {RegionEndpoint = AWSUtils.GetRegionEndpoint(this.service)});
             }
+            else if (!string.IsNullOrEmpty(this.profileName))
+            {
+                // AWS DynamoDB instance (auth via explicit credentials and token found in a named profile)
+                var chain = new CredentialProfileStoreChain();
+                if (chain.TryGetAWSCredentials(this.profileName, out var credentials))
+                {
+                    var immutableCredentials = credentials.GetCredentials();
+                    var sessionCredentials = new SessionAWSCredentials(immutableCredentials.AccessKey, immutableCredentials.SecretKey, immutableCredentials.Token);
+                    this.ddbClient = new AmazonDynamoDBClient(sessionCredentials, new AmazonDynamoDBConfig {RegionEndpoint = AWSUtils.GetRegionEndpoint(this.service)});
+                }
```

Comment:
Would it make sense to grab the credentials differently? Something similar to:
```csharp
CredentialProfile basicProfile;
AWSCredentials awsCredentials;
var sharedFile = new SharedCredentialsFile();
if (sharedFile.TryGetProfile("basic_profile", out basicProfile) &&
    AWSCredentialsFactory.TryGetAWSCredentials(basicProfile, sharedFile, out awsCredentials))
{
    using (var client = new AmazonS3Client(awsCredentials, basicProfile.Region))
    {
        var response = client.ListBuckets();
    }
}
```
from: https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html

I believe getting immutable credentials like that can result in a temporary token while other methods can be self updating.

Reply:
Thanks for the feedback, let me check that other PR.